packages/api/cms-api/src/blocks/table.block.ts

```diff
@@ -0,0 +1,138 @@
+import { BlockData, BlockField, BlockInput, createBlock, inputToData } from "@comet/blocks-api";
+import { plainToInstance, Type } from "class-transformer";
+import { IsArray, IsBoolean, IsEnum, IsNumber, IsString } from "class-validator";
+
+enum ColumnSize {
+    extraSmall = "extraSmall",
+    small = "small",
+    standard = "standard",
+    large = "large",
+    extraLarge = "extraLarge",
+}
+
+class TableBlockColumnData extends BlockData {
+    @BlockField()
+    id: string;
+
+    @BlockField()
+    position: number;
+
+    @BlockField({ type: "enum", enum: ColumnSize })
+    size: ColumnSize;
+
+    @BlockField()
+    highlighted: boolean;
+}
+
+class TableBlockColumnInput extends BlockInput {
+    @BlockField()
+    @IsString()
+    id: string;
+
+    @BlockField()
+    @IsNumber()
+    position: number;
+
+    @BlockField({ type: "enum", enum: ColumnSize })
+    @IsEnum(ColumnSize)
+    size: ColumnSize;
+
+    @BlockField()
+    @IsBoolean()
+    highlighted: boolean;
+
+    transformToBlockData(): TableBlockColumnData {
+        return inputToData(TableBlockColumnData, this);
+    }
+}
+
+class TableBlockRowColumnValueData extends BlockData {
+    @BlockField()
+    columnId: string;
+
+    @BlockField({ type: "string" })
+    value: string;
+}
+
+class TableBlockRowColumnValueInput extends BlockInput {
+    @BlockField()
+    @IsString()
+    columnId: string;
+
+    @BlockField({ type: "string" })
+    @IsString()
+    value: string;
+
+    transformToBlockData(): TableBlockRowColumnValueData {
+        return inputToData(TableBlockRowColumnValueData, this);
+    }
+}
+
+class TableBlockRowData extends BlockData {
+    @BlockField()
+    id: string;
+
+    @BlockField()
+    position: number;
+
+    @BlockField()
+    highlighted: boolean;
+
+    @BlockField(TableBlockRowColumnValueData)
+    @Type(() => TableBlockRowColumnValueData)
+    columnValues: TableBlockRowColumnValueData[] = [];
```

Comment:
Wouldn't cells be a better name?

Reply:
Done in 2b248cb05e0da069892384e241914b6850cbebd9